Render headings via Spectre.Console FigletText (IHeaderStyle, default H1) (#113)
boxofyellow/ConsoleMarkdownRenderer: README.md

| - | - | - | - |
| `Headers` | `List<IHeaderStyle>` | Used as overrides of `Header`, an order lists of styles to use for different level of headers | by default contains a single `FigletTextStyle` for H1; deeper levels fall back to `Header` |
| `WrapHeader` | `bool` | When `true`, will wrap Headers with `#`'s to denote the level (only applies to plain `TextStyle` heading entries; ignored for `FigletTextStyle`) | yes / `true` |
| `IncludeDebug` | `bool` | When `true` will display all content within in boxes to help visualize how the content is being interpreted by the tool | off / `false` |
| `ShowFencedCodeBlockInfo` | `bool` | When `true`, displays the info field (e.g., language identifier) from fenced code blocks | off / `false` |

### Rendering headings as FIGlet ASCII art

Each entry in `Headers` (and the `Header` fallback) is an [`IHeaderStyle`](https://github.com/boxofyellow/ConsoleMarkdownRenderer/blob/main/Styling/IHeaderStyle.cs).  Two implementations ship with the library:

- [`TextStyle`](https://github.com/boxofyellow/ConsoleMarkdownRenderer/blob/main/Styling/TextStyle.cs) — inline styled markup, optionally wrapped with `#` characters via `WrapHeader`.

> - [`FigletTextStyle`](https://github.com/boxofyellow/ConsoleMarkdownRenderer/blob/main/Styling/FigletTextStyle.cs) — renders the heading text as large ASCII art through Spectre.Console's [`FigletText`](https://spectreconsole.net/widgets/figlet) widget.  Exposes an optional `Justification` (`TextJustification.Left`, `Right`, or `Center`) and an optional `Foreground` color.  Instances are created via the static factory methods: `FigletTextStyle.Create(…)` for the default FIGlet font, and the async `FigletTextStyle.CreateAsync(fontPath, …)` to load a custom FIGlet font (`.flf`) which is read and cached once.  Because `FigletText` does not support decoration or background, `FigletTextStyle` is a peer of `TextStyle` (both implement `IHeaderStyle`) rather than a subclass.

Reviewer: ```suggestion
- [`FigletTextStyle`](https://github.com/boxofyellow/ConsoleMarkdownRenderer/blob/main/Styling/FigletTextStyle.cs) — renders the heading text as large ASCII art through Spectre.Console's [`FigletText`](https://spectreconsole.net/widgets/figlet) widget.  Exposes an optional `Justification` (`TextJustification.Left`, `Right`, or `Center`) and an optional `Foreground` color.  Instances are created via the static factory methods: `FigletTextStyle.Create(…)` for the default FIGlet font, and the async `FigletTextStyle.CreateAsync(fontPath, …)` to load a custom FIGlet font (`.flf`).  Because `FigletText` does not support decoration or background, `FigletTextStyle` is a peer of `TextStyle` (both implement `IHeaderStyle`) rather than a subclass.
```

**Default behavior.** By default the first entry in `Headers` is a `FigletTextStyle`, so top-level (`#`) headings render as FIGlet ASCII art out of the box.  Deeper levels (`##`, `###`, …) fall through to `Header` and continue to render as styled, `#`-wrapped markup.  To opt H1 to the inline styled markup behavior, replace `Headers[0]` (or clear `Headers`):

```csharp
var options = new DisplayOptions();

> options.Headers[0] = new TextStyle(decoration: TextDecoration.Bold); // H1 back to styled markup

Reviewer: @copilot I would suggest folks use `options.Headers.Clear()` to achieve this.

| name | type | description | default
| - | - | - | - |
| `Headers` | `List<IHeaderStyle>` | Used as overrides of `Header`, an order lists of styles to use for different level of headers | by default contains a single centered, blue `FigletTextStyle` for H1; deeper levels fall back to `Header` |
| `WrapHeader` | `bool` | When `true`, will wrap Headers with `#`'s to denote the level (only applies to plain `TextStyle` heading entries; ignored for `FigletTextStyle`) | yes / `true` |
| `IncludeDebug` | `bool` | When `true` will display all content within in boxes to help visualize how the content is being interpreted by the tool | off / `false` |
| `ShowFencedCodeBlockInfo` | `bool` | When `true`, displays the info field (e.g., language identifier) from fenced code blocks | off / `false` |

### Rendering headings as FIGlet ASCII art

Each entry in `Headers` (and the `Header` fallback) is an [`IHeaderStyle`](https://github.com/boxofyellow/ConsoleMarkdownRenderer/blob/main/Styling/IHeaderStyle.cs).  Two implementations ship with the library:

> - [`TextStyle`](https://github.com/boxofyellow/ConsoleMarkdownRenderer/blob/main/Styling/TextStyle.cs) — the original behavior: renders the heading as inline styled markup, optionally wrapped with `#` characters via `WrapHeader`.

Reviewer: @copilot this document should not really talk about history so let's not refer to this as "original behavior" how about this?

```suggestion
- [`TextStyle`](https://github.com/boxofyellow/ConsoleMarkdownRenderer/blob/main/Styling/TextStyle.cs) — inline styled markup, optionally wrapped with `#` characters via `WrapHeader`.
```

- [`FigletTextStyle`](https://github.com/boxofyellow/ConsoleMarkdownRenderer/blob/main/Styling/FigletTextStyle.cs) — renders the heading text as large ASCII art through Spectre.Console's [`FigletText`](https://spectreconsole.net/widgets/figlet) widget.  Exposes an optional `Justification` (`TextJustification.Left`, `Right`, or `Center`) and an optional `Foreground` color.  Because `FigletText` does not support decoration or background, `FigletTextStyle` is a peer of `TextStyle` (both implement `IHeaderStyle`) rather than a subclass.

> **Default behavior.** By default the first entry in `Headers` is a centered, blue `FigletTextStyle`, so top-level (`#`) headings render as FIGlet ASCII art out of the box.  Deeper levels (`##`, `###`, …) fall through to `Header` and continue to render as styled, `#`-wrapped markup.  To opt H1 back into the original styled-markup behavior, replace `Headers[0]` (or clear `Headers`):

Reviewer: @copilot how about this?
```suggestion
**Default behavior.** By default the first entry in `Headers` is a centered, blue `FigletTextStyle`, so top-level (`#`) headings render as FIGlet ASCII art out of the box.  Deeper levels (`##`, `###`, …) fall through to `Header` and continue to render as styled, `#`-wrapped markup.  To opt H1 to the inline styled markup behavior, replace `Headers[0]` (or clear `Headers`):
```

> ```csharp
> var options = new DisplayOptions();
> options.Headers[0] = new TextStyle(decoration: TextDecoration.Bold); // H1 back to styled markup
> ```
> 
> To opt a different level (or the `Header` fallback) in to FIGlet rendering, assign a `FigletTextStyle`:
> 
> ```csharp
> var options = new DisplayOptions();
> options.Headers.Add(new FigletTextStyle(
>     justification: TextJustification.Center,
>     foreground: TextColor.Blue));    // applies to '##' (H2)
> ```

Reviewer: @copilot this does not seem to be helper for this docuemnt
 
```suggestion
```

| `IncludeDebug` | `bool` | When `true` will display all content within in boxes to help visualize how the content is being interpreted by the tool | off / `false` |
| `ShowFencedCodeBlockInfo` | `bool` | When `true`, displays the info field (e.g., language identifier) from fenced code blocks | off / `false` |

> ### Rendering headings as FIGlet ASCII art

Reviewer: @copilot Much of this will need to be reworked when adding the new class relationship
